Signal/src/ViewControllers/MediaPageViewController.swift

```diff
@@ -654,3 +755,38 @@ class MediaPageViewController: UIPageViewController, UIPageViewControllerDataSou
         }
     }
 }
+
+class CaptionView: UIView {
+    var label: UILabel = UILabel()
+
+    var text: String? {
+        get { return label.text }
+        set { label.text = newValue }
+    }
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        let gradientView = GradientView(from: .clear, to: .black)
+        addSubview(gradientView)
+        gradientView.autoPinEdgesToSuperviewEdges()
+
+        addSubview(label)
+        label.font = UIFont.ows_dynamicTypeBody
+        label.textColor = .white
+
+        // Usually captions are short, but they can be as long as 2k.
+        // We don't have UI for viewing infinitely large captions, so
+        // we do some not-ideal things to broaden the lenght of the
```

Comment:
s/lenght/length/.

`minimumScaleFactor` a) reduces accessibility by rendering the text smaller, making it harder to read. b) still doesn't guarantee us that all text will fit in a reasonable way.  

What if instead we set `numberOfLines` to something like 5 and if users tap on the caption we push a simple view to display all of the text, e.g. `LongTextViewController`.  LGTV isn't pretty but it ensures that all of the caption is readable by all.

Reply:
Good point. Long captions are now in a scroll view. Getting a scroll view to grow to fit it's content, but only up to a point, was a little tricky.